File: python/src/test/QuesoManchegoTest.py
import unittest
from QuesoManchego import *

class QuesoManchegoTest(unittest.TestCase):
    def setUp(self):
        self.quesoManchego = QuesoManchego()

    def testCurrentTemperature(self):
        self.quesoManchego.setCurrentTemperature(21)
        self.assertEqual(21, self.quesoManchego.getCurrentTemperature())

    def testCurrentTemperature(self):
        self.quesoManchego.setCurrentTemperature(21)
        self.assertEqual(21, self.quesoManchego.getCurrentTemperature())

    def testFalseMelt(self):
        self.quesoManchego.melt(False)
        self.assertFalse(self.quesoManchego.isMelted())

    def testTrueMelt(self):
        self.quesoManchego.melt(True)
        self.assertTrue(self.quesoManchego.isMelted())

    def testMelting(self):
        self.assertEqual(20, self.quesoManchego.getMeltingTemperature())

File: csharp/unittest-practice/src/test/QuesoManchegoTest.cs
using NUnit.Framework;
using unittestpractice.main;

namespace unittestpractice.test
{
    [TestFixture]
    internal class QuesoManchegoTest
    {
        private QuesoManchego _quesoManchego;

        [SetUp]
        public void Setup()
        {
            _quesoManchego = new QuesoManchego();
        }

        [TestCase]
        public void TestCurrentTemperature()
        {
            _quesoManchego.SetCurrentTemperature(21);
            Assert.AreEqual(21, _quesoManchego.GetCurrentTemperature());
        }

        [TestCase]
        public void TestFalseMelt()
        {
            _quesoManchego.Melt(false);
            Assert.IsFalse(_quesoManchego.IsMelted());
        }

        [TestCase]
        public void TestTrueMelt()
        {
            _quesoManchego.Melt(true);
            Assert.IsTrue(_quesoManchego.IsMelted());
        }

        [TestCase]
        public void TestMelting()
        {
            Assert.AreEqual(10, _quesoManchego.GetMeltingTemperature());
        }
    }
}
